src/main/java/com/sphereon/factom/identity/did/IdentityFactory.java

```diff
@@ -21,12 +27,36 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 public class IdentityFactory {
     private static final IdAddressKeyOps ADDRESSES = new IdAddressKeyOps();
 
-    public IdentityResponse toIdentity(String identifier, List<FactomIdentityEntry<?>> entries) throws RuleException {
+    public BlockchainResponse<?> toBlockchainResponse(String identifier, List<FactomIdentityEntry<?>> entries) throws RuleException {
+        if (entries == null || entries.size() == 0) {
+            throw new RuleException("Identity for %s could not be resolved", identifier);
+        }
+        if (entries.get(0).getOperationValue().equals(OperationValue.IDENTITY_CHAIN_CREATION)) {
+            return toIdentity(identifier, entries);
+        }
+        else if (entries.get(0).getOperationValue().equals(OperationValue.DID_MANAGEMENT)) {
+            return toDidResponse(identifier, entries);
+        }
+        throw new RuleException("DID Chain for %s did not start with a valid external id.", identifier);
+    }
+
+
+    public DIDDocument toDid(String identifier, BlockchainResponse<?> blockchainResponse) throws URISyntaxException {
+        if(blockchainResponse.getContent() instanceof FactomDidContent){
```

Comment:
Suggest to create a private method which returns you the actual OperationValue enum from the entry very much like the check in toBlockchainResponse ( and thus to be used in the toBlockchainResponse as well).

Right now you basically first parse the first entry. Then cast it and then do an instanceOf check on the case to cast it to your target.  The last cast will still be there, but you would have a clean method returning the OperationValue constant for both methods to determine your response on

Reply:
I was also looking for a better way to do this, however in this method, I don't think there should be any reference to `OperationValue` as the `BlockchainResponse` doesn't wrap an entry, but rather the resolved DID information after processing all the entries (ie, either and `IdentityResponse` or a `FactomDidContent`). So, for instance, if the content is a `FactomDidContent` then all operations (`DID_MANAGEMENT`, `DID_UPDATE`. etc) have already been processed to form that DID Content. It's not clear to me that there should be any associations between an operation and the information inside this method.

Reply:
One solution would be to attach the OperationValue or type class to the Response. You can then either based on the enum or based on the class do the cast

Reply:
Yes, that's better. Updated.